sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -78,9 +78,42 @@ case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
     }
   }
 
-  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
```

Comment:
nested field is nothing special, just `CreateStruct` and `GetStructField` expressions. What we should handle is symmetric expressions, e.g. `CreateStruct` and `GetStructField`, `Add` and `Subtract`, etc.

It doesn't make sense to only handle `CreateStruct` and `GetStructField` specially, and leave others behind.